### PR TITLE
fix: Connector mgmt oas

### PR DESCRIPTION
**Pull request by dimakis**
## Description

Edit to connector_mgmt oas to pass linter.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~
~- [ ] Required Standard Operating Procedure (SOP) is added.~
~- [ ] JIRA has been created for changes required on the client side~

### CODE REVIEW

**Review by valdar — APPROVED** [review]

File: openapi/connector_mgmt.yaml [hunk 1]
```diff
@@ -1719,6 +1735,17 @@ components:
         error_handling:
           dead_letter_queue:
             topic: "dlq"
+        service_account:
+            id: "1"
+            kind: "ServiceAccount"
+            href: "/api/connector_mgmt/v1/serviceaccounts/1"
```
Comment (dimakis) [inline openapi/connector_mgmt.yaml:1741]: @valdar openapi spec looks for a service account here. currently there isn't any endpoint for `/api/connector_mgmt/v1/serviceaccounts/1` I'm hesitant to populate with a non functional example. I'm unsure what way to proceed though, do you have any opinion?

**Review by dhirajsb — APPROVED** [review]
lgtm

**Comment by dhirajsb** [comment]
@dimakis is this change supposed to go in before the coming code freeze?

**Comment by machi1990** [comment]
The change depends on https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1130 being merged first.

**Comment by dimakis** [comment]
this rebase is a mess. don't anyone waste their time on this until I have that sorted correctly.

**Comment by dimakis** [comment]
@machi1990 I'm now seeing his fail integration tests on what I believe is the updated Errors format. I'm not 100% though. Does this mean that the tests need major refactoring do you think?

**Comment by machi1990** [comment]
> @machi1990 I'm now seeing his fail integration tests on what I believe is the updated Errors format. I'm not 100% though. Does this mean that the tests need major refactoring do you think?

I think the issue is precisely coming from https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1138/files#diff-79f0404acc4bc801897a86872b31dd083bdc8bbade622fa43a85ec284c8198d5R1003 
i.e since the fields are marked as required, they won't be omitted when marshalling in case they are not set i.e when they are empty string.
The connector team might help with the desired behavior here /cc @dhirajsb @valdar @lburgazzoli

**Comment by dimakis** [comment]
@dhirajsb LGTM, thank you

@lburgazzoli would you mind having a look at this please.

**Comment by dimakis** [comment]
@JameelB I believe you're a member of the reviewers, if you get a chance could yo have a look?

**Review by JameelB — APPROVED** [review]

**Comment by wtrocki** [comment]
Thank you all for approving PR. Myself and @dimakis missing merge permission. Looking to get that integrated into main

**Pull request merged**